tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -2261,276 +2259,6 @@ private[emitter] class JSDesugaring(semantics: Semantics,
-      (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9')
-
-    def containsOnlyValidChars(): Boolean = {
-      val len = module.length
-      var i = 0
-      while (i != len) {
-        if (!isValidChar(module.charAt(i)))
-          return false
-        i += 1
-      }
-      true
-    }
-
-    def buildValidName(): String = {
-      val result = new java.lang.StringBuilder("$i_")
-      val len = module.length
-      var i = 0
-      while (i != len) {
-        val c = module.charAt(i)
-        if (isValidChar(c))
-          result.append(c)
-        else
-          result.append("$%04x".format(c.toInt))
-        i += 1
-      }
-      result.toString()
-    }
-
-    val varName =
-      if (containsOnlyValidChars()) "$i_" + module
-      else buildValidName()
-
-    js.VarRef(js.Ident(varName, Some(module)))
-  }
-
-  private[emitter] def envField(field: String, subField: String,
-      origName: Option[String] = None)(
-      implicit pos: Position): js.Tree = {
-    import TreeDSL._
-
-    outputMode match {
-      case OutputMode.ECMAScript51Global =>
-        envField(field) DOT js.Ident(subField, origName)
-
-      case OutputMode.ECMAScript51Isolated | OutputMode.ECMAScript6 =>
-        js.VarRef(js.Ident("$" + field + "_" + subField, origName))
-    }
-  }
-
-  private[emitter] def envField(field: String)(
-      implicit pos: Position): js.Tree = {
-    import TreeDSL._
-
-    outputMode match {
-      case OutputMode.ECMAScript51Global =>
-        js.VarRef(js.Ident(ScalaJSEnvironmentName)) DOT field
-
-      case OutputMode.ECMAScript51Isolated | OutputMode.ECMAScript6 =>
-        js.VarRef(js.Ident("$" + field))
-    }
-  }
-
-  private[emitter] def genPropSelect(qual: js.Tree, item: js.PropertyName)(
-      implicit pos: Position): js.Tree = {
-    item match {
-      case item: js.Ident         => js.DotSelect(qual, item)
-      case item: js.StringLiteral => genBracketSelect(qual, item)
-      case js.ComputedName(tree)  => genBracketSelect(qual, tree)
-    }
-  }
-
-  private[emitter] def genBracketSelect(qual: js.Tree, item: js.Tree)(
-      implicit pos: Position): js.Tree = {
-    item match {
-      case js.StringLiteral(name) if internalOptions.optimizeBracketSelects &&
-          isValidIdentifier(name) && name != "eval" =>
-        /* We exclude "eval" because Rhino does not respect the strict mode
-         * specificities of eval().
-         */
-        js.DotSelect(qual, js.Ident(name))
-      case _ =>
-        js.BracketSelect(qual, item)
-    }
-  }
-
-  private[emitter] def genIdentBracketSelect(qual: js.Tree, item: String)(
-      implicit pos: Position): js.Tree = {
-    require(item != "eval")
-    if (internalOptions.optimizeBracketSelects)
-      js.DotSelect(qual, js.Ident(item))
-    else
-      js.BracketSelect(qual, js.StringLiteral(item))
-  }
-
   class DesugarException(tree: Tree,
       cause: Throwable) extends Exception(exceptionMsg(tree), cause)
```

Comment:
I think this one is completely unused. Candidate for a removal in the "Remove unused stuff" commit?